src/classes/HiFiMixerSession.ts

```diff
@@ -583,9 +682,52 @@ export class HiFiMixerSession {
      * and when a user can hear other users in a Space due to the fact that the Bluetooth audio device must
      * switch I/O profiles.
      * - Not all browsers support the `navigator.permissions` API
+     * 
+     * @param newMutedValue If `true`, the input audio stream will be muted. If `false`, the input stream will be unmuted.
+     * @param tryToStopMicStream If `false`, this function will use Method 2 described above to mute or unmute the input audio stream. If `true`, this function will use Method 1.
      * @returns `true` if the stream was successfully muted/unmuted, `false` if it was not.
      */
     async setInputAudioMuted(newMutedValue: boolean, tryToStopMicStream: boolean = false): Promise<boolean> {
+        return await this._setMuted(newMutedValue, tryToStopMicStream, MuteReason.CLIENT);
+    }
+
+    async _setMutedByAdmin(mutedByAdmin: boolean, muteReason: MuteReason): Promise<boolean> {
+        // For now:
+        // - Admin muting should mute the client, and prevent the client from unmuting
+        // - Admin unmuting should not unmute the client, but simply allow the client to unmute
+        // - When the connection ends, the client is allowed to unmute, which for now is equivalent to an admin unmute
+        this._adminPreventsInputAudioUnmuting = mutedByAdmin;
+        return await this._setMuted(mutedByAdmin || this._lastSuccessfulInputAudioMutedValue, false, muteReason);
+    }
+
+    async _setMuted(newMutedValue: boolean, tryToStopMicStream: boolean, muteReason: MuteReason): Promise<boolean> {
+        let success = true;
+        if (muteReason == MuteReason.CLIENT) {
+            if (this._adminPreventsInputAudioUnmuting && !newMutedValue) {
+                HiFiLogger.warn(`Couldn't set mute state: Muted by admin.`);
```

Comment:
Is this also managed on the mixer side? For instance, if User A gets muted by an admin and tries to unmute themself, the `_adminPreventsInputAudioUnmuting` flag prevents the unmuting from happening here. However, if User A is a l33t hax0r, and manages to change `_adminPreventsInputAudioUnmuting` to `false`, could they then unmute themselves? Or does the mixer say "Nope, not unmuting, admin said so."

Reply:
_adminPreventsInputAudioUnmuting is set to `true` when the client is muted on the mixer side, i.e. even if they were l33t hax0r and set it to `false` on the javascript side, it would do them no good as the mixer would continue ignoring their audio.